project/ws/app/src/lib/routes/user-profile/components/request-dialog/request-dialog.component.ts

```diff
@@ -0,0 +1,139 @@
+import { Component, Inject, OnInit } from '@angular/core'
+import { FormGroup, FormControl, Validators, AbstractControl, ValidatorFn } from '@angular/forms'
+import { MAT_DIALOG_DATA, MatDialogRef, MatSnackBar } from '@angular/material'
+// tslint:disable-next-line: import-name
+import _ from 'lodash'
+import { Observable } from 'rxjs'
+
+import { v4 as uuid } from 'uuid'
+import { RequestService } from 'src/app/routes/public/public-request/request.service'
+
+export function forbiddenNamesValidatorPosition(optionsArray: any): ValidatorFn {
+  return (control: AbstractControl): { [key: string]: any } | null => {
+    if (!optionsArray) {
+      return null
+      // tslint:disable-next-line: no-else-after-return
+    } else {
+      const index = optionsArray.findIndex((op: any) => {
+        // tslint:disable-next-line: prefer-template
+        // return new RegExp('^' + op.channel + '$').test(control.channel)
+        return op.name === control.value.name
+      })
+      return index < 0 ? { forbiddenNames: { value: control.value.name } } : null
+    }
+  }
+}
+@Component({
+  selector: 'ws-app-request-dialog',
+  templateUrl: './request-dialog.component.html',
+  styleUrls: ['./request-dialog.component.scss'],
+})
+export class RequestDialogComponent implements OnInit {
+  requestForm!: FormGroup
+  namePatern = `[a-zA-Z\\s\\']{1,32}$`
+  customCharsPattern = `^[a-zA-Z0-9 \\w\-\&\(\)]*$`
+  customCharsPatternOrg = `^[a-zA-Z0-9 \\w\-\&,\(\)]*$`
+  confirm = false
+  disableBtn = false
+  requestType: any
+  masterPositions!: Observable<any> | undefined
+  userData: any
+
+  // tslint:disable-next-line:max-line-length
+  requestObj: {
+    state: string; action: string; serviceName: string; userId: string;
+    applicationId: string; actorUserId: string; deptName: string; updateFieldValues: any
+  } | undefined
+  formobj: { toValue: {}; fieldKey: any; description: any; firstName: any; email: any; mobile: any } | undefined
+  userform: any
+
+  constructor(
+    private snackBar: MatSnackBar,
+    private requestSvc: RequestService,
+    private dialogRef: MatDialogRef<RequestDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) private data: any
+  ) {
+    this.requestType = this.data.reqType
+    this.userData = this.data
+
+    this.requestForm = new FormGroup({
+      // tslint:disable-next-line:max-line-length
+      designation: new FormControl('', this.requestType === 'Position' ? [Validators.pattern(this.customCharsPattern),
```

Comment:
Is Position we are using  or coming from api?

Reply:
It is coming from queryparams